### PR TITLE
feat(m-extensions): Adds divu and remu

### PR DESCRIPTION
**Pull request by mw2000**
We are now adding DIVU and REMU instructions as part of the M-Extensions

### CODE REVIEW

File: jolt-core/src/jolt/instruction/divu.rs [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                imm: None,
+                virtual_sequence_index: Some(0),
+            },
+            register_state: RegisterState {
+                rs1_val: None,
+                rs2_val: None,
+                rd_post_val: Some(q),
+            },
+            memory_state: None,
+            advice_value: Some(23434), // What should advice value be here?
```
Comment (moodlezoup) [inline jolt-core/src/jolt/instruction/divu.rs:47]: > What should advice value be here?

We know `x` and `y`, so we can just compute the quotient and remainder:

```rust
let q = x / y;
let r = x - q * y;
```

And `q` and `r` are the values we use for these two ADVICE instructions

Comment (mw2000) [reply jolt-core/src/jolt/instruction/divu.rs]: Got it, added!

File: jolt-core/src/jolt/trace/rv.rs [hunk 2]
```diff
@@ -145,7 +145,8 @@ impl TryFrom<&RVTraceRow> for RV32I {
             RV32IM::MULHU => Ok(MULHUInstruction(row.register_state.rs1_val.unwrap(), row.register_state.rs2_val.unwrap()).into()),
 
             // TODO: Input of Advice is not from register
-            RV32IM::VIRTUAL_ADVICE => Ok(ADVICEInstruction(row.register_state.rs1_val.unwrap()).into()),
+            RV32IM::VIRTUAL_ADVICE => Ok(ADVICEInstruction(row.advice_value.unwrap()).into()),
+            // TODO: Actually move value from one operand to another
```
Comment (moodlezoup) [inline jolt-core/src/jolt/trace/rv.rs:149]: we can delete these TODOs now I think

File: common/src/rv_trace.rs
Comment (moodlezoup) [inline common/src/rv_trace.rs:1]: we can delete the `MOVE` stuff in this file

File: jolt-core/src/jolt/instruction/divu.rs [hunk 3]
```diff
@@ -0,0 +1,259 @@
+use common::constants::virtual_register_index;
+use tracer::{ELFInstruction, RVTraceRow, RegisterState, RV32IM};
+
+use super::VirtualInstructionSequence;
+use crate::jolt::instruction::{
+    add::ADDInstruction, beq::BEQInstruction, mulu::MULUInstruction, sltu::SLTUInstruction,
+    virtual_advice::ADVICEInstruction, virtual_assert_lte::ASSERTLTEInstruction, JoltInstruction,
+};
+/// Perform signed*unsigned multiplication and return the upper WORD_SIZE bits
```
Comment (moodlezoup) [inline jolt-core/src/jolt/instruction/divu.rs:9]: nit: update comment

File: jolt-core/src/jolt/instruction/divu.rs [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+                imm: None,
+                virtual_sequence_index: Some(0),
+            },
+            register_state: RegisterState {
+                rs1_val: None,
+                rs2_val: None,
+                rd_post_val: Some(q),
+            },
+            memory_state: None,
+            advice_value: Some(quotient), // What should advice value be here?
```
Comment (moodlezoup) [inline jolt-core/src/jolt/instruction/divu.rs:48]:
```suggestion
            advice_value: Some(quotient),
```

File: jolt-core/src/jolt/instruction/divu.rs [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+                imm: None,
+                virtual_sequence_index: Some(1),
+            },
+            register_state: RegisterState {
+                rs1_val: None,
+                rs2_val: None,
+                rd_post_val: Some(r),
+            },
+            memory_state: None,
+            advice_value: Some(remainder), // What should advice value be here?
```
Comment (moodlezoup) [inline jolt-core/src/jolt/instruction/divu.rs:68]:
```suggestion
            advice_value: Some(remainder),
```

File: jolt-core/src/jolt/instruction/remu.rs [hunk 6]
```diff
@@ -0,0 +1,262 @@
+use common::constants::virtual_register_index;
+use tracer::{ELFInstruction, RVTraceRow, RegisterState, RV32IM};
+
+use super::VirtualInstructionSequence;
+use crate::jolt::instruction::{
+    add::ADDInstruction, beq::BEQInstruction, mulu::MULUInstruction, sltu::SLTUInstruction,
+    virtual_advice::ADVICEInstruction, virtual_assert_lte::ASSERTLTEInstruction, JoltInstruction,
+};
+
+/// Perform signed*unsigned multiplication and return the upper WORD_SIZE bits
```
Comment (moodlezoup) [inline jolt-core/src/jolt/instruction/remu.rs:10]: nit: fix comment

**Review by moodlezoup — COMMENTED** [review]
Nice work! Just a few cosmetic nits

**Review by moodlezoup — APPROVED** [review]
Thanks!

**Pull request merged**